exonum/src/runtime/types.rs

```diff
@@ -148,12 +203,18 @@ impl FromStr for ArtifactId {
     }
 }
 
+/// Exhaustive service instance specification.
 #[derive(Debug, Clone, PartialEq, Eq, Hash, ProtobufConvert, Serialize, Deserialize)]
 #[exonum(pb = "schema::runtime::InstanceSpec", crate = "crate")]
 pub struct InstanceSpec {
+    /// Unique service instance identifier.
     pub id: ServiceInstanceId,
-    pub artifact: ArtifactId,
+    /// Unique service instance name.
+    ///
+    /// It should contains only these characters: `a-zA-Z0-9` and one of `_-./`.
```

Comment:
contain

Reply:
Are slashes allowed? I considered using it as the path prefix, if slashes were allowed, they would have to be escaped 🤔